### PR TITLE
Data generation

### PR DESCRIPTION
**Pull request by Daniellebenba**
can you approve pleaseeeeeeee WOW

### CODE REVIEW

**Comment by guyfreund** [comment]
> can you approve pleaseeeeeeee WOW

@Daniellebenba no 😄

**Comment by Daniellebenba** [comment]
....

**Comment by Daniellebenba** [comment]
and now

File: src/pipeline/data_drift_detection/detectors/statistical_based_detector.py [hunk 1]
```diff
@@ -14,6 +14,8 @@
 from src.pipeline.config import Config
 
 
+# TODO: add detector for new unseen values of categorical features
```
Comment (guyfreund) [inline src/pipeline/data_drift_detection/detectors/statistical_based_detector.py:17]: nice

File: src/pipeline/data_generation/data_generation_manager.py [hunk 2]
```diff
@@ -1,13 +1,66 @@
...
+    def manage(self) -> DataDrift:
+        is_drifted = np.random.choice([False, True])
+        generated_data = self._get_generated_dataset(is_drifted)
+        #TODO: do saving dataset call from utils function
+        return DataDrift(is_drifted=is_drifted)
+
+    def _get_generated_dataset(self, is_drifted: bool) -> Union[np.array, pd.DataFrame]:
+        if is_drifted:
+            drift_types_list = np.random.choice(self._data_drift_types,
+                                                size=np.random.randint(1, len(self._data_drift_types) + 1))
```
Comment (guyfreund) [inline src/pipeline/data_generation/data_generation_manager.py:51]: not sure I understand this line

File: src/pipeline/datasets/deployment_datasets.py [hunk 3]
```diff
@@ -64,3 +64,6 @@ def __init__(self):
 
     def load(self) -> pd.DataFrame:
         raise NotImplementedError  # TODO: implement
+
+
+
```
Comment (guyfreund) [inline src/pipeline/datasets/deployment_datasets.py:69]:
```suggestion

```
delete

**Review by guyfreund — CHANGES_REQUESTED** [review]
Nice.
- [x] Let's go over the requirements together once they are in their finalized version
- [x] Let's discuss the change in the data generator interface

**Pull request merged**